src/main/io/ledstrip.c

```diff
@@ -303,80 +283,70 @@ bool parseLedStripConfig(int ledIndex, const char *config)
             config++;   // skip separator
         }
         switch (parseState) {
-            case X_COORDINATE:
-                x = atoi(chunk);
-                break;
-            case Y_COORDINATE:
-                y = atoi(chunk);
-                break;
-            case DIRECTIONS:
-                for (char *ch = chunk; *ch; ch++) {
-                    for (ledDirectionId_e dir = 0; dir < LED_DIRECTION_COUNT; dir++) {
-                        if (directionCodes[dir] == *ch) {
-                            direction_flags |= LED_FLAG_DIRECTION(dir);
-                            break;
-                        }
+        case X_COORDINATE:
+            x = atoi(chunk);
+            break;
+        case Y_COORDINATE:
+            y = atoi(chunk);
+            break;
+        case DIRECTIONS:
+            for (char *ch = chunk; *ch; ch++) {
+                for (ledDirectionId_e dir = 0; dir < LED_DIRECTION_COUNT; dir++) {
+                    if (directionCodes[dir] == *ch) {
+                        direction_flags |= LED_FLAG_DIRECTION(dir);
+                        break;
                     }
                 }
-                break;
-            case FUNCTIONS:
-                for (char *ch = chunk; *ch; ch++) {
-                    for (ledBaseFunctionId_e fn = 0; fn < LED_BASEFUNCTION_COUNT; fn++) {
-                        if (baseFunctionCodes[fn] == *ch) {
-                            baseFunction = fn;
-                            break;
-                        }
+            }
+            break;
+        case FUNCTIONS:
+            for (char *ch = chunk; *ch; ch++) {
+                for (ledBaseFunctionId_e fn = 0; fn < LED_BASEFUNCTION_COUNT; fn++) {
+                    if (baseFunctionCodes[fn] == *ch) {
+                        baseFunction = fn;
+                        break;
                     }
-
-                    for (ledOverlayId_e ol = 0; ol < LED_OVERLAY_COUNT; ol++) {
-                        if (overlayCodes[ol] == *ch) {
-                            overlay_flags |= LED_FLAG_OVERLAY(ol);
-                            break;
-                        }
+                }
+                for (ledOverlayId_e ol = 0; ol < LED_OVERLAY_COUNT; ol++) {
+                    if (overlayCodes[ol] == *ch) {
+                        overlay_flags |= LED_FLAG_OVERLAY(ol);
+                        break;
                     }
                 }
-                break;
-            case RING_COLORS:
-                color = atoi(chunk);
-                if (color >= LED_CONFIGURABLE_COLOR_COUNT)
-                    color = 0;
-                break;
-            case PARSE_STATE_COUNT:; // prevent warning
+            }
+            break;
+        case RING_COLORS:
+            color = atoi(chunk);
```

Comment:
Codacy found an issue: [Unless checked, the resulting number can exceed the expected range (CWE-190).  If source untrusted, check both minimum and maximum, even if the input had no minus sign (large numbers can roll over into negative number; consider saving to an unsigned value if that is intended). ](https://app.codacy.com/gh/emuflight/EmuFlight/pullRequest?prid=6326343)